### ISSUE TITLE
upgrade from 3.3.4 to 3.3.5 results in glyphicons do not display correct

### ISSUE THREAD

**Issue by AllanHoejgaardJensen** [twbs/bootstrap-sass#927]
I just upgraded from _3.3.4_ to _3.3.5_ and glyphicons did not display correctly after the upgrade. Downgrading to 3.3.4 solves the problem. It seems to find the glyphicons but for some reason they are not displayed anymore.

The notes for 3.3.5 does not say that changes are necessary for the upgrade to 3.3.5.

_Is anyone else having this issue with 3.3.5?_

**Comment by glebm** [comment]
Hi, what version of Sass are you using?

**Comment by Bochenski** [comment]
I've had the same issue.  Seems to be related to the @at-root directive which is leaving the following in my outputted css
````
@at-root {
  @font-face {
    font-family: 'Glyphicons Halflings';
    src: url('../fonts/glyphicons-halflings-regular.eot');
    src: url('../fonts/glyphicons-halflings-regular.eot?#iefix') format('embedded-opentype'), url('../fonts/glyphicons-halflings-regular.woff2') format('woff2'), url('../fonts/glyphicons-halflings-regular.woff') format('woff'), url('../fonts/glyphicons-halflings-regular.ttf') format('truetype'), url('../fonts/glyphicons-halflings-regular.svg#glyphicons_halflingsregular') format('svg'); }
 }
````

possibly because I'm using node-sass (and possibly an out of date version of that)?

**Comment by cyrilpic** [comment]
Setting the following dependency for sass-rails forces the update of sass gem:

`gem 'sass-rails', '>= 5.0.0'`

sass-rails versions 4.0.1 to 4.0.5 do not accept sass 3.3.0

**Comment by glebm** [comment]
v3.3.5.1 of the ruby gem comes with the updated Sass dependency (no other changes). For node-sass, please upgrade to the latests version.

**Comment by guillaumevincent** [comment]
if you need to upgrade node-sass to the latest version, just type

    npm install --save-dev gulp-sass

**Comment by dvddmt** [comment]
I have the same issue with @at-root. Here is my Gemfile.lock
```
source 'http://rubygems.org'
gem 'roo'
gem 'nokogiri', '1.6.6.2'

gem 'rails', '4.1.1'

gem 'bootstrap-sass', '~> 3.3.5.1'
gem 'sass-rails', '>= 5.0.0'
gem 'sass', '~> 3.4.12'
gem 'uglifier', '>= 1.3.0'
gem 'coffee-rails', '~> 4.0.0'


gem 'rspec-rails', '2.13.1'
group :test do
  gem 'selenium-webdriver', '2.35.1'
  gem 'capybara', '2.1.0'
end
gem 'jquery-rails'
gem 'turbolinks'
gem 'jbuilder', '~> 2.0'
gem 'yaml_db', :git => 'git://github.com/ludicast/yaml_db.git'

  gem 'bcrypt-ruby', '~> 3.1.2', :require => 'bcrypt'
  
gem 'will_paginate', '3.0.4'
gem 'bootstrap-will_paginate', '0.0.9'

gem 'sdoc', '~> 0.4.0',          group: :doc

gem 'tzinfo-data', platforms: [:mingw, :mswin]
group :production do

  gem 'rails_12factor', '0.0.2'
end
gem 'rubyXL'
gem 'therubyracer', :platforms => :ruby
 gem 'sprockets-rails', '>= 1.0.0', '< 2.1'
gem "activesupport","4.1.1"
gem 'thin'
		gem 'sqlite3'
# Use Unicorn
platforms :ruby do # linux
gem 'unicorn'
end
# Use PostgreSQL
gem 'pg', group: :production

group :development do
  # Use Capistrano for deployment
  gem 'capistrano', '~> 3.1'
  gem 'capistrano-rails', '~> 1.1.1'
  gem 'capistrano-bundler'
  gem 'capistrano-rbenv', '~> 2.0'
  gem 'capistrano-unicorn-nginx', '~> 2.0'
  gem 'capistrano-postgresql', '~> 3.0'
end
```
And in my application.scss i have
```
@import "bootstrap-sprockets";
@import "bootstrap";
```

**Comment by fabiode** [comment]
Bumping up sass-rails to >= 5.0.0 in my Gemfile solved the problem.
Shouldn't the README file on master be:

```ruby
gem 'bootstrap-sass', '~> 3.3.5'
gem 'sass-rails', '>= 5.0.0'
```
?

**Comment by glebm** [comment]
One can use sass with Rails without using sass-rails. I think the issue is that it depends on sass 3.3.0+ but `@at-root` only works correctly since 3.3.4+ according to http://sass-lang.com/documentation/file.SASS_CHANGELOG.html#334_21_march_2014

**Comment by dvddmt** [comment]
Thanks all, I've solved the problem by deleting cache. 
```
rake tmp:cache:clear  
rake assets:clean
```

**Comment by aguegu** [comment]
Building with gulp, I solved this by upgrading gulp-sass from "^1.3.3" to "^2.0.4", like black magic ...

**Comment by danielpclark** [comment]
Thanks for your tip @dvddmt.  I had to revert back to

```ruby
gem 'bootstrap-sass', '~> 3.3.4.1'
```

and do the rake clean commands.
